App crashes if unknown host is entered

**Actual Behaviour**

The app crashes if the user enters an invalid host address.

**Expected Behaviour**

The app never crashes due to user input.

**Steps to reproduce it**

1. Start app
2. Choose "Connect Device" in the Navigation Drawer
3. Tap "WIFI" button
4. Enter ".lo"
5. Tap "CONNECT" button

**LogCat for the issue**
```
FATAL EXCEPTION: AsyncTask #2
Process: io.pslab, PID: 26126
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
	at java.util.concurrent.FutureTask.run(FutureTask.java:269)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.IllegalArgumentException: Invalid URL host: ".lo"
	at okhttp3.HttpUrl$Builder.parse$okhttp(HttpUrl.kt:1338)
	at okhttp3.HttpUrl$Companion.get(HttpUrl.kt:1634)
	at okhttp3.Request$Builder.url(Request.kt:184)
	at io.pslab.fragment.ESPFragment$ESPTask.doInBackground(ESPFragment.java:101)
	at io.pslab.fragment.ESPFragment$ESPTask.doInBackground(ESPFragment.java:87)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305) 
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644) 
	at java.lang.Thread.run(Thread.java:1012) 
```

**Screenshots of the issue**

<img src="https://github.com/user-attachments/assets/6807477b-1aa1-435e-a6b4-7ff9b808609e" width="400" />

**Would you like to work on the issue?**

Yes.

--- a/app/src/main/java/io/pslab/fragment/ESPFragment.java
+++ b/app/src/main/java/io/pslab/fragment/ESPFragment.java
@@ -54,7 +54,6 @@ public class ESPFragment extends DialogFragment {
             if (espIPAddress.isEmpty() && ((activity = getActivity()) != null)) {
                 CustomSnackBar.showSnackBar(activity.findViewById(android.R.id.content),
                         getString(R.string.incorrect_IP_address_message), null, null, Snackbar.LENGTH_SHORT);
-
             } else {
                 new ESPTask().execute();
             }
@@ -116,7 +115,7 @@ public class ESPFragment extends DialogFragment {
                 CustomSnackBar.showSnackBar(activity.findViewById(android.R.id.content),
                         getString(R.string.incorrect_IP_address_message), null, null, Snackbar.LENGTH_SHORT);
             } else {
-                Log.v("Response", result);
+                Log.v(TAG, "Response: " + result);
             }
         }
     }

--- a/app/src/main/java/io/pslab/fragment/ESPFragment.java
+++ b/app/src/main/java/io/pslab/fragment/ESPFragment.java
@@ -4,15 +4,14 @@ import android.app.Activity;
 import android.os.AsyncTask;
 import android.os.Bundle;
 import android.util.Log;
-import android.view.KeyEvent;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.view.WindowManager;
 import android.view.inputmethod.EditorInfo;
 import android.widget.Button;
 import android.widget.EditText;
 import android.widget.ProgressBar;
-import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
@@ -30,6 +29,8 @@ import okhttp3.Request;
 import okhttp3.Response;
 
 public class ESPFragment extends DialogFragment {
+    private static final String TAG = ESPFragment.class.getSimpleName();
+
     private String espIPAddress = "";
     private ProgressBar espConnectProgressBar;
     private Button espConnectBtn;
@@ -46,31 +47,24 @@ public class ESPFragment extends DialogFragment {
         EditText espIPEditText = rootView.findViewById(R.id.esp_ip_edit_text);
         espConnectBtn = rootView.findViewById(R.id.esp_connect_btn);
         espConnectProgressBar = rootView.findViewById(R.id.esp_connect_progressbar);
-        espConnectBtn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                espIPAddress = espIPEditText.getText().toString();
-                espConnectBtn.onEditorAction(EditorInfo.IME_ACTION_DONE);
-                Activity activity;
-                if (espIPAddress.isEmpty() && ((activity = getActivity()) != null)) {
-                    CustomSnackBar.showSnackBar(activity.findViewById(android.R.id.content),
-                            getString(R.string.incorrect_IP_address_message), null, null, Snackbar.LENGTH_SHORT);
+        espConnectBtn.setOnClickListener(v -> {
+            espIPAddress = espIPEditText.getText().toString();
+            espConnectBtn.onEditorAction(EditorInfo.IME_ACTION_DONE);
+            Activity activity;
+            if (espIPAddress.isEmpty() && ((activity = getActivity()) != null)) {
+                CustomSnackBar.showSnackBar(activity.findViewById(android.R.id.content),
+                        getString(R.string.incorrect_IP_address_message), null, null, Snackbar.LENGTH_SHORT);
 
-                } else {
-                    new ESPTask().execute();
-                }
+            } else {
+                new ESPTask().execute();
             }
-
         });
-        espIPEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
-            @Override
-            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
-                if (actionId == EditorInfo.IME_ACTION_DONE) {
-                    espConnectBtn.performClick();
-                    return true;
-                }
-                return false;
+        espIPEditText.setOnEditorActionListener((v, actionId, event) -> {
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                espConnectBtn.performClick();
+                return true;
             }
+            return false;
         });
         return rootView;
     }
@@ -78,10 +72,10 @@ public class ESPFragment extends DialogFragment {
     @Override
     public void onResume() {
         super.onResume();
-        ViewGroup.LayoutParams params = getDialog().getWindow().getAttributes();
+        WindowManager.LayoutParams params = getDialog().getWindow().getAttributes();
         params.height = ViewGroup.LayoutParams.WRAP_CONTENT;
         params.width = ViewGroup.LayoutParams.MATCH_PARENT;
-        getDialog().getWindow().setAttributes((android.view.WindowManager.LayoutParams) params);
+        getDialog().getWindow().setAttributes(params);
     }
 
     private class ESPTask extends AsyncTask<Void, Void, String> {
@@ -100,14 +94,15 @@ public class ESPFragment extends DialogFragment {
                 Request request = new Request.Builder()
                         .url("http://" + espIPAddress)
                         .build();
-                Response response = client.newCall(request).execute();
-                if (response.code() == 200) {
-                    ScienceLabCommon.setIsWifiConnected(true);
-                    ScienceLabCommon.setEspBaseIP(espIPAddress);
+                try (Response response = client.newCall(request).execute()) {
+                    if (response.code() == 200) {
+                        ScienceLabCommon.setIsWifiConnected(true);
+                        ScienceLabCommon.setEspBaseIP(espIPAddress);
+                    }
+                    result = response.body().string();
                 }
-                result = response.body().string();
-            } catch (IOException e) {
-                e.printStackTrace();
+            } catch (IllegalArgumentException | IOException e) {
+                Log.e(TAG, "Unable to get data from " + espIPAddress, e);
             }
             return result;
         }

